### DISCUSSION TITLE
Patient count disappears from My Saved Queries

### DISCUSSION THREAD

**Discussion by lawdank** [uwrit/leaf#608] [Q&A]
Hello Leaf Folks,

Under Saved Queries -> My Saved Queries, Patient Count is not visible (screenshot below):
<img width="1047" alt="image" src="https://github.com/user-attachments/assets/cb2444d8-f4f8-4fcf-b0fe-a4c4e79ffaef">
For the same user as Admin, when I navigate to User Saved Queries I can see the Patient  Count:
<img width="1075" alt="image" src="https://github.com/user-attachments/assets/59c348b6-ede2-4fbf-9ea9-f9278b3f506e">

Is this a bug or some issue with stored procedure? 
We are using Leaf Version - 3.11.1

Thanks,
Sheersha

**Comment by ndobb** [comment]
Thanks @lawdank, let me dig a bit and get back to you later this week.

**Comment by lawdank** [comment]
Hi @ndobb, 

Thank you! Did you get a chance to look into this?


Thanks,
Sheersha

**Reply by ndobb** [reply]
I did, @lawdank, but am puzzled as I can't recreate the issue, either in dev or on servers Leaf (3.11.1) is deployed to. I've experimented with changing permissions as well as reviewing source code.

In your browser, can you please open up the developer tools -> Network tab, then repeat the process to get your first screenshot?

In the Network tab you should see something like:
![image](https://github.com/user-attachments/assets/f5bcfaa6-cc26-4896-a7fc-6b7b095b340d)
with a "query" API request.

If you inspect the response of that (for first or whatever entry), do you see a `count` property with data?
![image](https://github.com/user-attachments/assets/958974e8-80de-4ac8-ad84-4f95c5e77a0c)

That should at least help us isolate whether this is a server- or client-side issue.

**Comment by lawdank** [comment]
Hi @ndobb,

Thank you for the details. I checked the response & do not see count property for query API request. 
<img width="981" alt="Pasted Graphic" src="https://github.com/user-attachments/assets/07874fc0-c82b-40aa-a675-198b335fcdf9">
When I check for a specific user the count property is available.
<img width="775" alt="image" src="https://github.com/user-attachments/assets/a7d1a32b-4eab-48aa-8863-9a25958a9af0">

**Comment by lawdank** [comment]
Hi @ndobb,

Would you have an idea on how to debug this further?

**Comment by ndobb** [comment]
@lawdank! Thanks for your patience.

I spent some time debugging this but after stepping back a moment the reason became clear. In your appsettings.json file, are you enabling [`Cohort.Noise`](https://github.com/uwrit/leaf/blob/master/src/server/API/appsettings.json#L104) (a random +/- value to each query count) or [`Cohort.LowCellSizeMasking`](https://github.com/uwrit/leaf/blob/master/src/server/API/appsettings.json#L109) (not showing true counts below a threshold)?

If yes then that is the reason. If users are never to know the true count, then showing such a count in the Saved Queries would be a potentially problematic workaround. The Leaf source code [explicitly checks and removes the count if either of those cases are true](https://github.com/uwrit/leaf/blob/master/src/server/Model/Search/QueryManager.cs#L41). I wrote the code for this years ago and it took a bit to recall.

**Comment by lawdank** [comment]
Hi @ndobb,

I think that is it. We have  [Cohort.LowCellSizeMasking](https://github.com/uwrit/leaf/blob/master/src/server/API/appsettings.json#L109) enabled.  Thank you!

A few researchers have raised queries that they would like the ability to view count of saved queries. It is time consuming for them currently. And we cannot turn off Cohort.LowCellSizeMasking at our end. 
Would you consider an implementation where we have the option in appsettings.json that allows us to control count visibility for saved queries?

**Reply by ndobb** [reply]
Hi Sheersha,

I'm not sure about this one - it would be a blatant exploit users could execute to view actual counts, and defeats the purpose of enabling low cell size masking in the first place. You and team can of course do as you see fit with the code, but from my perspective I can't imagine merging such a change in - it would likely be a major issue for other sites deploying Leaf if they were caught unawares.

If you could provide an alternative step-by-step explanation of how you would imagine this working without revealing the actual patient count, I would be happy to consider it (and implement myself if it made sense for other sites too).

Best,
-nic
